glusterd2/peer/self.go

```diff
@@ -39,18 +40,27 @@ func normalizeAddrs() ([]string, error) {
 
 // AddSelfDetails results in the peer adding its own details into etcd
 func AddSelfDetails() error {
-	var err error
 
+	var err error
 	p := &Peer{
 		ID:            gdctx.MyUUID,
 		Name:          gdctx.HostName,
 		PeerAddresses: []string{config.GetString("peeraddress")},
 	}
-
 	p.ClientAddresses, err = normalizeAddrs()
 	if err != nil {
 		return err
 	}
 
+	peerInfo, err := GetPeer(gdctx.MyUUID.String())
```

Comment:
Why these changes required?

Reply:
In case the glusterd2 service was restarted and the data (metadata ) already existed then it is important to go and get back the data .